### PR TITLE
Feature email payment

### PR DESCRIPTION
**Pull request by BoAnd**
Working code. I know, I'm awesome. This should send a email when there is made a payment.

### CODE REVIEW

File: stregsystem/utils.py [hunk 1]
```diff
@@ -72,3 +73,13 @@ def date_to_midnight(date):
     :return: the date as a timezone aware datetime at midnight
     """
     return timezone.make_aware(timezone.datetime(date.year, date.month, date.day, 0, 0))
+
+
+def send_mail(user_email):
+    msg = "Your money is now on your streg account"
```
Comment (VirtualSatai) [inline stregsystem/utils.py]: Could we improve the message? 
f"Hey {member.firstname}, we've added {payment.amount} to your streg-account: '{member.username}'"

Is there an subject field? to set?

File: stregsystem/utils.py [hunk 2]
```diff
@@ -72,3 +73,13 @@ def date_to_midnight(date):
...
+    try:
+       smtpObj = smtplib.SMTP('localhost')
+       smtpObj.sendmail('FIT@fklub.dk', user_email, msg)
+    except:
+       pass
```
Comment (VirtualSatai) [inline stregsystem/utils.py]: Logging of mistakes?

File: stregsystem/utils.py (hunk 1 above)
Comment (VirtualSatai) [reply stregsystem/utils.py]: Maybe we could add: "If you don't want to receive more spam contact: fklub@elefsennet.dk"

Comment (BoAnd) [reply stregsystem/utils.py]: Yes, we should improve the message. And I like your ideas.

File: stregsystem/utils.py (hunk 2 above)
Comment (BoAnd) [reply stregsystem/utils.py]: Yes, but where to ?

File: stregsystem/models.py [hunk 3]
```diff
@@ -304,7 +304,7 @@ def save(self, *args, **kwargs):
             self.member.make_payment(self.amount)
             super(Payment, self).save(*args, **kwargs)
             self.member.save()
-            send_mail(self.member.email)            
+            send_mail(self.member, self.amount)
```
Comment (VirtualSatai) [inline stregsystem/models.py]: Add a condition that so that we only try to send an email if the fember has one.

File: stregsystem/utils.py [hunk 4]
```diff
@@ -75,8 +75,8 @@ def date_to_midnight(date):
     return timezone.make_aware(timezone.datetime(date.year, date.month, date.day, 0, 0))
 
 
-def send_mail(user_email):
-    msg = "Your money is now on your streg account"
+def send_mail(member, payment):
+    msg = "Hey {member.firstname} we've added {payment.amount} to your streg-account: '{member.username}'"
     try:
        smtpObj = smtplib.SMTP('localhost')
        smtpObj.sendmail('FIT@fklub.dk', user_email, msg)
```
Comment (VirtualSatai) [inline stregsystem/utils.py]: `user_email` should be `member.email` <- if that variable exists, otherwise exit?

File: stregsystem/utils.py (hunk 2 above)
Comment (DelusionalLogic) [reply stregsystem/utils.py]: To here: https://docs.djangoproject.com/en/2.0/topics/logging/

File: stregsystem/utils.py [hunk 5]
```diff
@@ -72,3 +73,13 @@ def date_to_midnight(date):
     :return: the date as a timezone aware datetime at midnight
     """
     return timezone.make_aware(timezone.datetime(date.year, date.month, date.day, 0, 0))
+
+
+def send_mail(member, amount):
```
Comment (DelusionalLogic) [inline stregsystem/utils.py]: This doesn't look like a generic send_mail function. Maybe it should be named something like "send_payment_mail"?

File: stregsystem/utils.py [hunk 6]
```diff
@@ -72,3 +73,13 @@ def date_to_midnight(date):
     :return: the date as a timezone aware datetime at midnight
     """
     return timezone.make_aware(timezone.datetime(date.year, date.month, date.day, 0, 0))
+
+
+def send_payment_mail(member, amount):
+    msg = f"Hey {member.firstname} we've added {amount} to your streg-account: '{member.username}'"
```
Comment (VirtualSatai) [inline stregsystem/utils.py]: Maybe prepend a subject field:
https://stackoverflow.com/questions/7232088/python-subject-not-shown-when-sending-email-using-smtplib-module?answertab=votes#tab-top

File: stregsystem/utils.py [hunk 7]
```diff
@@ -76,10 +76,27 @@ def date_to_midnight(date):
...
+    msg['Subject'] = 'Streg-account payment'
+    
+    html = f"
+    <html>
+        <head></head>
+        <body>
+            <p>Hey {member.firstname}<br>
+               We've added {amount}$ to your streg-account: {member.username}<br>
+               <br>
+               If you don't want to receive more spam contact: fklub@elefsennet.dk
```
Comment (VirtualSatai) [inline stregsystem/utils.py]: `<a href="mailto:fklub@elefsennet.dk?Subject=Klage" target="_top">fklub@elefsennet.dk</a>`

**Pull request merged**